src/extension.ts

```diff
@@ -1,176 +1,208 @@
+  const themeList = await getThemeList(extensionConfig, userSettings);
 
   const currentThemeName: string | undefined = userSettings.get('workbench.colorTheme');
 
   if (typeof currentThemeName !== 'undefined') {
-    saveThemes(themeList.concat(currentThemeName).sort(), 'addedtheme', currentThemeName);
+    saveThemes(themeList.concat(currentThemeName).sort(), Messages.AddedTheme(currentThemeName));
   }
 }
 
-function removeCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+async function removeCurrentTheme(extensionConfig = getExtensionConfig(), userSettings = getUserSettings()) {
+  const themeList = await getThemeList(extensionConfig, userSettings);
 
   const currentThemeName: string | undefined = userSettings.get('workbench.colorTheme');
 
   if (typeof currentThemeName !== 'undefined') {
-    saveThemes(themeList.filter(th => th !== currentThemeName), 'removedtheme', currentThemeName);
+    saveThemes(themeList.filter(th => th !== currentThemeName), Messages.RemovedTheme(currentThemeName));
   }
 }
 
-export async function activate(context: vscode.ExtensionContext) {
-  const switchTheme = vscode.commands.registerCommand('randomThemeSwitcher.switchTheme', () => {
-    changeTheme();
-  });
-
-  context.subscriptions.push(switchTheme);
-
-  const copyThemesToSettings = vscode.commands.registerCommand('randomThemeSwitcher.copyInstalledThemes', () => {
-    const installedThemes = getInstalledThemes();
-
-    saveThemes(installedThemes, 'copyall');
-  });
-
-  context.subscriptions.push(copyThemesToSettings);
-
-  const addCurrentThemeToSettogs = vscode.commands.registerCommand('randomThemeSwitcher.addCurrentTheme', () => {
-    addCurrentTheme();
-  });
-
-  context.subscriptions.push(addCurrentThemeToSettogs);
-
-  const removeCurrentThemeToSettogs = vscode.commands.registerCommand('randomThemeSwitcher.removeCurrentTheme', () => {
-    removeCurrentTheme();
-  });
+let switchMode: SwitchModes = 'manual';
+let isSwitching = false;
+let themeList: string[] = new Array();
 
-  context.subscriptions.push(removeCurrentThemeToSettogs);
+export async function activate(context: vscode.ExtensionContext) {
 
-  const extensionConfig = getExtensionConfig();
+  let extensionConfig = getExtensionConfig();
+
+  context.subscriptions.push(
+    vscode.commands.registerCommand(CommandsIds.Switch, () => {
+      changeTheme();
+    }),
+    vscode.commands.registerCommand(CommandsIds.CopyAll, async () => {
+      const installedThemes = await getInstalledThemes();
+
+      saveThemes(installedThemes, Messages.CopiedTheme(installedThemes.length));
+    }),
+    vscode.commands.registerCommand(CommandsIds.Add, () => {
+      addCurrentTheme();
+    }),
+    vscode.commands.registerCommand(CommandsIds.Remove, () => {
+      removeCurrentTheme();
+    }),
+
+    // it listen for theme manual switches, then update the LAST_THEME_MATERIAL settings.
```

Comment:
ottimo, non ci avevo pensato. Bravo

Reply:
Grazie 😊✌🏻

Ho paura che l'issue #10 sia collegato a questo: è possibile che quando passa su un tema Material lui tenga il passaggio e non cambia la volta dopo.

Stavo pensando ad un miglioramento su questo lato: si potrebbe indicare una lista (tipo la material list che hai già costruito) come opzione che fa in modo di tenere il tema ancora una volta se questo è compreso.
Altra storia sarebbe quella delle probabilità:
creare una lista dei preferiti(starred) che fa sii che i temi listati vengano proposti con maggior frequenza (quello sarebbe figo).

Però vedi tu, 
per me è già ottima così !

Reply:
Potrebbe essere una idea, direi di aprire due issue al riguardo così non ci perdiamo questa cosa

Reply:
Ho fatto prima a farlo 😂
Solo che adesso non so come procedere: ti apro una nuova PR, va bene ?

Per la seconda ok la apro 👍🏻

Reply:
Ho aperto una PR per il primo e un issue per il secondo 👍🏻